### PR TITLE
Promoting the dynamics fitter out of 'experimental' status

### PR DESCRIPTION
**Pull request by keenon**
This changes both the frontend and `engine.py` to promote subjects to run the dynamics fitter by default, unless users explicitly opt out.

We're also removing the "my model has no arms" option on the frontend, since we don't need that anymore (the processing code will auto-detect), and reformatting the dynamics fitter options to be inline with other options, instead of set aside in an "experimental" card.

### CODE REVIEW

File: frontend/src/pages/mocap/MocapSubjectView.tsx [hunk 1]
```diff
@@ -886,9 +886,8 @@ const MocapSubjectView = observer((props: MocapSubjectViewProps) => {
   let exportSDF = props.cursor.subjectJson.getAttribute("exportSDF", false);
   let exportMJCF = props.cursor.subjectJson.getAttribute("exportMJCF", false);
   let ignoreJointLimits = props.cursor.subjectJson.getAttribute("ignoreJointLimits", false);
-  let fitDynamics = props.cursor.subjectJson.getAttribute("fitDynamics", false);
+  let dontFitDynamics = props.cursor.subjectJson.getAttribute("dontFitDynamics", false);
```
Comment (nickbianco) [inline frontend/src/pages/mocap/MocapSubjectView.tsx:889]: Perhaps `ignoreDynamics` or `skipDynamicsFitting` instead.

File: server/engine/engine.py [hunk 2]
```diff
@@ -602,7 +609,8 @@ def processLocalSubjectFolder(path: str, outputName: str = None, href: str = '')
                     # Specifically optimize to 0-ish residuals, if user requests it
                     if residualsToZero:
                         for trial in range(len(dynamicsInit.poseTrials)):
-                            originalTrajectory = dynamicsInit.poseTrials[trial].copy()
+                            originalTrajectory = dynamicsInit.poseTrials[trial].copy(
+                            )
```
Comment (nickbianco) [inline server/engine/engine.py:613]: This is probably auto-formatting, but looks weird to me.

File: server/engine/engine.py [hunk 3]
```diff
@@ -1187,20 +1190,25 @@ def processLocalSubjectFolder(path: str, outputName: str = None, href: str = '')
             if fitDynamics and dynamicsInit is not None:
                 residualForce = trialProcessingResult['linearResidual']
                 residualTorque = trialProcessingResult['angularResidual']
-                f.write(f'  - Avg. Residual Force   = {residualForce:1.2f} N\n')
-                f.write(f'  - Avg. Residual Torque  = {residualTorque:1.2f} N-m\n')
+                f.write(
+                    f'  - Avg. Residual Force   = {residualForce:1.2f} N\n')
+                f.write(
+                    f'  - Avg. Residual Torque  = {residualTorque:1.2f} N-m\n')
```
Comment (nickbianco) [inline server/engine/engine.py:1196]: I normally don't care about the auto-formatting too much, but a lot of these changes are adding unnecessary lines. If it's easy to revert, great. If not, don't worry about it.

**Review by nickbianco — CHANGES_REQUESTED** [review]
Mostly looks good. Suggesting a variable rename and perhaps revert some of the auto-formatting.

File: server/engine/engine.py (hunk 2 above)
Comment (keenon) [reply server/engine/engine.py:613]: Yes, I can't figure out how to convince my auto-formatter to suck less... :/ I'll go fix this manually.

**Comment by AlbertoCasasOrtiz** [comment]
I jus tested this by reprocessing my trial, this time with dynamics fit (I let `Skip Dynamics Fit` unchecked) and it is still not showing the list of variables to plot. I checked the Logs and 'fitDynamics' is set to false.

**Comment by keenon** [comment]
> I jus tested this by reprocessing my trial, this time with dynamics fit (I let `Skip Dynamics Fit` unchecked) and it is still not showing the list of variables to plot. I checked the Logs and 'fitDynamics' is set to false.
> 
> I've tried preprocessing again, this time checking `Skip Dynamics Fit`, and in the Logs 'fitDynamics' is set to false again.

This is because we haven't redeployed the dev servers with the backend changes here yet. These buttons will currently have no effect on the backend.

**Review by nickbianco — APPROVED** [review]
LGTM.

**Pull request merged**